zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -975,6 +975,9 @@ public void restart(String settingId, String user, String noteId) throws Interpr
 
   // restart in note page
   public void restart(String settingId, ExecutionContext executionContext) throws InterpreterException {
+    if (settingId == null || settingId.startsWith("__internal__")) {
```

Comment:
How about adding comment to describe `__internal__`?
```suggestion
    // __internal__ is set by RunNotebookInterpreter
    if (settingId == null || settingId.startsWith("__internal__")) {
```